src/store/movieStore.ts

```diff
@@ -15,6 +15,13 @@ const useMovieStore = defineStore("movieStore", () => {
     setFilterParam,
     searchMovies,
   } = useSearch(initialList.value);
+  
+  const fetchMovies = async () => {
+    const { movies, error, getAllMovies } = useMovies();
+    await getAllMovies();
+    if(error.value) initialList.value = mockMoviesList;
```

Comment:
It's quite confusing that your Promise responding with positive data if some error appears. You also should use reject in this case

Reply:
fixed